dap-dart.el

```diff
@@ -0,0 +1,105 @@
+;;; dap-dart.el --- Debug Adapter Protocol mode for Dart language -*- lexical-binding: t; -*-
```

Comment:
I can merge it it as it is but IMO this should go in lsp-dart project considering the following points:

1. It will be easier for the users to configure everything - fewer things to install if you are trying to make emacs dart IDE.
2. You will own it and it will be easier for you to change it to add new templates.
3. You will reference it either way from lsp-dart in order to implement debug lenses.

Reply:
I see... I think we I can create a `lsp-dart-dap.el` on `lsp-dart`, what do you think?

Reply:
It is up to you to decide for the name. Having the adapter in lsp-dart.el or keeping the current name dap-dart.el is fine too.

Reply:
@yyoncho, I'll move tihis file to `lsp-dart`, but can we merge the `dap-utils` change?